GLMakie/assets/shader/lines.frag

```diff
@@ -10,6 +10,7 @@ in vec4 f_color;
 in vec2 f_uv;
 in float f_thickness;
 flat in uvec2 f_id;
+flat in int f_type; // TODO bad for performance
```

Comment:
Given the comment in the scatter shader

https://github.com/MakieOrg/Makie.jl/blob/4dab9955103282b6af93e5bc516a9797a5b8273f/GLMakie/assets/shader/distance_shape.frag#L26

I assume we should find a way around this?

Reply:
Did some benchmark with

```julia
let 
    GLMakie.closeall()
    ps = [2 * rand(Point2f) .- 1 for _ in 1:2000]

    function myrenderloop(screen)
        yield()
        GLMakie.GLFW.SwapInterval(0)
        GLMakie.pollevents(screen)
        GLMakie.render_frame(screen)
        GLMakie.GLFW.SwapBuffers(GLMakie.to_native(screen))
        yield()

        @time for _ in 1:10_000
            GLMakie.pollevents(screen)
            GLMakie.render_frame(screen)
            GLMakie.GLFW.SwapBuffers(GLMakie.to_native(screen))
        end

        while GLMakie.isopen(screen) && !screen.stop_renderloop
            GLMakie.pollevents(screen)
            sleep(0.1)
            yield()
        end

        return
    end

    scene = Scene()
    linesegments!(scene, ps, linewidth = 10)
    screen = GLMakie.Screen(renderloop = myrenderloop)
    display(screen, scene)
end
```

I get the same ~5% CI got for this branch vs master. 

I tried hacking together a two pass version which does the old lines in the first pass and caps in the second pass. This way we can swap out the `flat in int f_type` for `uniform int linecap` and `uniform int render_pass`. This ends up being about 10% slower than master though, doing both passes. Doing just the line pass is 3-4% slower.

Reply:
I think that's ok for linecaps! Can we optionally disable it and get the 5% back for high perf use cases? We could also think about having a different recipe for that though 🤷 I think the default should make good lines, and performance should be the special case!

Reply:
Reducing the number of options in the fragment shader ([5e206a9](https://github.com/MakieOrg/Makie.jl/pull/2536/commits/5e206a98e258097a1dbddb98374ed17a98175a3d)) helped a bit ... I think. I'm never sure with gpu benchmarking. When I tested it I got a bunch of benchmarks that were ~2% slower than master and then a bunch up to 5% slower :shrug:

Reply:
I ran them again without background processes and I'm getting ~2% now ([812ff3f](https://github.com/MakieOrg/Makie.jl/pull/2536/commits/812ff3f1691da3c255c0c56cc1c173f17c1b333d))